### PR TITLE
[Windows] Fixes UDS integration & caches socket domain

### PR DESCRIPTION
**Pull request by davinci26**
Commit Message:

The PR aims to fix UDS integration tests on Windows. To that end the following changes were made:

1. Fix `domain` function usage on Windows by caching the domain from the socket creation and passing down to the io_handler.
1. Adds the correct length for UDS on Windows.
1. Enables the test. 

Risk Level: Low
Testing: Existing tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by davinci26** [comment]
cc: @envoyproxy/windows-dev

File: source/common/network/io_socket_handle_impl.h
Comment (davinci26) [inline source/common/network/io_socket_handle_impl.h:21]: @wrowe @sunjayBhatia @mattklein123 this is the place that I intended to use the `platform` enum.

This could have been written with a `constexpr if (current_platform == platformTypes::Windows` instead of `#ifdefs`.

File: source/common/network/io_socket_handle_impl.h
Comment (davinci26) [inline source/common/network/io_socket_handle_impl.h:35]: This should probably not be defaulted and it should be a required parameter.

**Comment by wrowe** [comment]
/wait

For defaults resolution

**Comment by davinci26** [comment]
@wrowe marking it as a draft to query some full builds in the CI to figure out which consumers are setting the default value for `domain` and see why this is the case

File: source/common/network/io_socket_handle_impl.h
Comment (sunjayBhatia) [reply source/common/network/io_socket_handle_impl.h:21]: Seeing this example, and other situations it could be applied, I could see this pattern working, or just pulling this into a define in platform.h

Since this function is only used in one place, might be worth moving into the file it is used in an anonymous namespace like helper functions are in other places

File: source/common/network/io_socket_handle_impl.h
Comment (davinci26) [reply source/common/network/io_socket_handle_impl.h:35]: Since many tests inject fake fds, I kept the same signature and added `absl::nullopt` as the default

File: source/common/network/io_socket_handle_impl.h
Comment (davinci26) [reply source/common/network/io_socket_handle_impl.h]: I moved the helper function into `io_socket_handle_impl.cc` in an anonymous namespace in 783c656ed4c27cb06f855164af1765b648fe706c.

The idea that I had in mind was to have the platform.h for types and the rest to be moved near the place that they are used.

**Comment by davinci26** [comment]
@wrowe i expect the ci to go green and it should be ready for review

**Comment by davinci26** [comment]
`clang-tidy` fails due to https://github.com/envoyproxy/envoy/issues/13162 that is coming from upstream

**Review by wrowe — DISMISSED** [review]
LGTM, thanks!

**Comment by davinci26** [comment]
@envoyproxy/senior-maintainers can I also get a review from you.

File: source/common/network/io_socket_handle_impl.cc
Comment (mattklein123) [inline source/common/network/io_socket_handle_impl.cc:217]: I think this is a mistake. Same below. Revert?

File: source/common/network/io_socket_handle_impl.h
Comment (mattklein123) [inline source/common/network/io_socket_handle_impl.h:89]: nit: const

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with small comments, thanks.

/wait

File: source/common/network/io_socket_handle_impl.cc
Comment (davinci26) [reply source/common/network/io_socket_handle_impl.cc:217]: you mean the comment `// namespace Network`?

Comment (davinci26) [reply source/common/network/io_socket_handle_impl.cc:217]: and the comments `// namespace <name>` in general in the file

File: source/common/network/io_socket_handle_impl.h
Comment (davinci26) [reply source/common/network/io_socket_handle_impl.h:89]: thanks! will change

File: source/common/network/io_socket_handle_impl.cc
Comment (mattklein123) [reply source/common/network/io_socket_handle_impl.cc:217]: This comment is on the trailing } of a function which I think is an accident.

Comment (davinci26) [reply source/common/network/io_socket_handle_impl.cc:217]: I see thanks! I think my `clang-format` messed up the parsing and added them and I never realized it.

File: source/common/network/io_socket_handle_impl.h
Comment (davinci26) [reply source/common/network/io_socket_handle_impl.h]: fixed in c1d4d36567caae65137fe36f66fe188c0622533c

File: source/common/network/io_socket_handle_impl.cc
Comment (davinci26) [reply source/common/network/io_socket_handle_impl.cc]: fixed in c1d4d36567caae65137fe36f66fe188c0622533c

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Comment by davinci26** [comment]
Looks like `clang-format` is not happy with the new changes. My working hypothesis is that it has some trouble with the following functions

```
absl::optional<uint32_t> maybeGetPacketsDroppedFromHeader(
#ifdef SO_RXQ_OVFL
    const cmsghdr& cmsg) {
  if (cmsg.cmsg_type == SO_RXQ_OVFL) {
    return *reinterpret_cast<const uint32_t*>(CMSG_DATA(&cmsg));
  }
#else
    const cmsghdr&) {
#endif
  return absl::nullopt;
}
```
and `Address::InstanceConstSharedPtr maybeGetDstAddressFromHeader(const cmsghdr& cmsg,
                                                             uint32_t self_port, os_fd_t fd) {`

working in simplifying the `#ifeds`

**Comment by mattklein123** [comment]
If it forces it and is a bug in the formatter don't worry about it.

**Comment by davinci26** [comment]
Actually it is easy to fix if you simplify the functions above so it might be an overall win. I just transformed the `#ifdefs` into some `constexpr` functions. 

Verifying my changes and pushing another commit

**Comment by davinci26** [comment]
WDYT of the changes in 509d1c3f83636dbf08988845bda561f7a411347f? Locally it fixes my `clang-format`.

**Comment by davinci26** [comment]
@mattklein123 the CI all green again. feel free to take a look at the last 2 commits when you get some time.

**Review by mattklein123 — APPROVED** [review]
Thanks, nice cleanup!

**Pull request merged**
